lib/Language/Haskell/Stylish.hs

```diff
@@ -113,8 +120,17 @@ format maybeConfigPath maybeFilePath contents = do
 
 --------------------------------------------------------------------------------
 -- | Searches Haskell source files in any given folder recursively.
-findHaskellFiles :: Bool -> [FilePath] -> IO [FilePath]
-findHaskellFiles v fs = mapM (findFilesR v) fs >>= return . concat
+findHaskellFiles :: Bool -> [FilePath] -> ExcludeList -> IO [FilePath]
+findHaskellFiles v fs es =
+  mapM (findFilesR v) (excludeFiles fs es) >>= return . flip excludeFiles es . concat
```

Comment:
The reason to call `excludeFiles`
- _before_ performing directory traversing is to eliminate folders that are explicitly specified to be excluded, so recursing on them would be a waste of work anyway. E.g. large folders.
and
- _after_, so any explicitly specified file is to be excluded in a recursively traversed folder. E.g.: `stylish-haskell -r -e ./src/Bambi.hs ./src`